module/VuFind/src/VuFind/OAuth2/Entity/AuthCodeEntity.php

```diff
@@ -68,8 +68,8 @@ public function jsonSerialize(): mixed
         foreach ($properties as $property) {
             $result[$property] = $this->{$property};
         }
-        $client = $this->getClient();
-        $result['clientIdentifier'] = $client ? $client->getIdentifier() : null;
+        // Only call getClient() if we know the client has been initialized:
```

Comment:
```suggestion
        // Only call getClient() if we know the client has been initialized to avoid trouble with its return type:
```

Reply:
Not sure if this is better than just `$this->client?->getIdentifier()` with a comment about getClient()'s return type, but I suppose there could in theory be some logic in getClient().